### PR TITLE
fix: show failed list loads instead of an empty result

### LINKED ISSUE #672: Cannot distinguish between no documents for filter or offline after filter modification

### LINKED ISSUE #692: Surface interactive list-fill failures in Recently browsed mode
Parent tracker: #656 (item 37).

## Problem

If an interactive list fill fails in *Recently browsed* mode, the cache correctly remains incomplete and a later open retries it, but the user receives no explanation for why the visible list stopped loading.

The existing query-sync-error surface is limited to *Entire library* mode and identifies errors by saved-view name; it cannot accurately represent an ad-hoc filter. This differs from #663, which concerns background sync failures.

### Concrete reproduction: changing filters or saved views while the server is unreachable

1. Open the document list.
2. Make the configured server unreachable.
3. Change the active filter or switch to another saved view.
4. The document list becomes empty without an error indication.

The empty state misleadingly suggests that no documents match the selected query, rather than explaining that its results could not be loaded. The cache mode for this reported reproduction has not yet been established; verify it in both *Recently browsed* and *Entire library* modes and cover whichever modes are affected.

Server-unreachable versus device-offline error classification is tracked separately in #667; this issue owns showing the failed interactive load in the active list.

## Desired UX

- Surface failure in the list the user is actively viewing.
- Distinguish the default view, a saved view, and an ad-hoc query accurately.
- Distinguish unavailable query results from a successful query with no matching documents.
- Provide an appropriate retry path.
- Do not report an incomplete fill as complete.

## Acceptance criteria

- A failed interactive fill produces visible, correctly scoped feedback in *Recently browsed* mode.
- Changing a filter or saved view while the server is unreachable does not silently present an empty list as a successful result with no matches.
- Verify that reproduction in both cache modes and handle each affected mode.
- A successful query with zero matches still shows the normal empty state.
- Retry succeeds without requiring an app restart.
- Background-only failures remain governed by #663.

### PR DESCRIPTION
**Pull request by paulgessinger**
### Show failed list loads instead of an empty result

Changing the filter or switching saved views while the server is unreachable left the document list showing "No documents", indistinguishable from a query with zero matches.

**Why.** A filter change calls `DocumentListViewModel.refresh(filter:)` without `userInitiated`: it points the list's observation at the new query, then calls `fillQuery`. Offline, page 1 fails before anything is written, and the error was only logged, since `surface()` stays silent for non-user-initiated refreshes. With no rows, not fetching and no count, the view fell through to `NoDocumentsView`. In *Recently browsed* this hits any query not opened before; a later-page failure also stopped the list partway, silently. In *Entire library* the default list and saved views are pre-filled, so they correctly show cached rows offline, but ad-hoc filters and saved views whose library fill failed went empty the same way. The existing `QuerySyncError` surface is Offline & Sync only and keyed by saved-view name, so it could not describe either case.

**Now.**
- `Persistence`: `QueryStatus` reports whether the cached query is complete: the `filled_at` stamp plus an order that still reaches the server's total, so a list cut by the *Recently browsed* cap or downgrade reads incomplete. (The last position is compared rather than the row count, since a complete fill can have gaps from page-boundary repeats or remote deletes.)
- `DataModel`: a pure document-list state derived from cached rows, completeness and the last fill outcome, with tests.
- The list uses it:
  - *failed, nothing cached* → an error state titled for the default list, the saved view by name, or "this filter" (an edited saved view counts as a filter), with the error's existing description and a Retry button; pull-to-refresh works too.
  - *failed, partial cache* → rows stay, with an inline banner above them and Retry.
  - *failed, complete cache* → no banner; offline browsing of a fully cached list is unchanged.
  - *zero matches* → the normal empty state, including a zero-match result cached from an earlier full fill.

Retry re-runs only the fill, so placeholders appear immediately rather than after an element sync that may time out. When the list already shows the failure, pull-to-refresh doesn't also raise a toast for it. A filter change now switches the observed query before the element sync, so the previous query's rows or error don't linger meanwhile. Failures from a fill the list itself superseded are ignored. When another fill (e.g. the library sweep) takes the list's query over, the list follows it (`CachingBackend.waitForQueryWriters`) and, if the cache is still incomplete once nothing is filling it, shows the same banner / error state. The decision lives in `DataModel` (`DocumentListFillTracking`, with tests). Background-only failures stay with #663; error wording is refined separately by #667.

Closes #692. Closes #672 (same symptom; covered by the same state). Part of #656 (item 37).

### Manual test checklist

- [x] *Recently browsed*: open the list, make the server unreachable (or use the debug force-offline switch), apply a never-used filter → error state with filter wording. Switch to an unopened saved view → its name is in the title. Restore the server, tap Retry → rows load without restarting.
- [x] *Entire library*, offline: switch saved views → cached rows, no banner. Apply an ad-hoc filter → error state.
- [x] Online, a filter matching nothing → normal "No documents".
- [x] Large uncached view on a slow link, cut the connection after page 1 → rows plus banner; Retry once back online.
- [x] Banner and error state on iPad split view and with large Dynamic Type.
- [x] *Entire library*: open the default list on a large library, trigger a foreground library sweep, cut the connection mid-sweep → rows plus banner (or the error state if nothing is cached), not a silent partial list or endless placeholders.
- [x] *Entire library* → *Recently browsed* downgrade, then go offline and refresh the default list → the capped rows show the incomplete banner.
- [x] Redesign the top bar that's scrolled under. Screenshot 2026-09-13 at 15 21 50

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_0137Gp2EU4pSXez1ApezqnZk

### CODE REVIEW

**Comment by paulgessinger** [comment]
@codex

File: Persistence/Sources/Persistence/Database+Documents.swift [hunk 1]
```diff
@@ -514,7 +514,7 @@ extension Database {
       .fetchCount(db)
     return QueryStatus(
       totalCount: meta?.totalCount, localCount: localCount,
-      orderStale: meta?.orderStale ?? false)
+      orderStale: meta?.orderStale ?? false, isComplete: meta?.filledAt != nil)
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Persistence/Sources/Persistence/Database+Documents.swift]: 🤖 _Posted by Claude on Paul's behalf._

Confirmed, fixed in e66372c5.

`truncateQuery` keeps `filled_at` on purpose, and that has to stay: `QueryRetention.mostRecentlyFilled` ranks cached queries by the stamp, so clearing it would get a capped list collected by the reachability sweep (the recurring Recently-browsed cap in #722 relies on this). So the fix is in how completeness is read, not in the truncation.

`QueryStatus.isComplete` now needs the stamp **and** an order that reaches the server's total: `MAX(position) + 1 >= total_count`. I used the last position rather than the suggested `localCount >= totalCount` because a complete fill can hold fewer rows than the total. The `(server_id, query_key, remote_id)` unique key is `ON CONFLICT IGNORE`, so a document repeated across a page boundary is skipped and leaves a gap, and a remote delete prunes a row. Positions still run to the end in both cases. A truncation always removes the tail, so the last position drops below the total. Edge cases:
- no recorded total: the stamp decides;
- total 0 with an empty order: complete;
- more positions than the total (it shrank mid-fill): complete;
- every row cut with a positive total: incomplete;
- a remote delete of the *last* row: incomplete until the next fill. That's the conservative direction, and it only shows when a fill fails.

**Verified:** `swift test --package-path Persistence` passes (180 tests). New tests in `DowngradeGCTests`:
- a filled query truncated by `truncateQueryOrder` is no longer complete but keeps its stamp and total;
- `reclaimAfterDowngrade` leaves the capped default list incomplete, while a cap larger than the list keeps it complete;
- a complete fill with a skipped page-boundary repeat and a mid-list remote delete still reads complete;
- a table of the rule's edge cases.

The existing `observeQueryStatus` completeness test had stamped a 1-of-2 order as complete, so it now writes page 2 first.

**Inferred, not tested:** that #722's cap stays correct on top of this. I read its diff: it calls the same `truncateQuery` and keeps `filled_at`, so a capped list now reads incomplete there as well. I didn't build the two branches together.

File: swift-paperless/ViewModel/DocumentListViewModel.swift [hunk 2]
```diff
@@ -182,20 +263,62 @@ class DocumentListViewModel {
...
+  private func watchCompletion(of handle: QueryFillHandle, generation: Int) {
+    completionTask?.cancel()
+    completionTask = Task { @MainActor [weak self] in
+      do {
+        try await handle.awaitCompletion()
+      } catch {
+        // Cancellation is the list moving on, or another fill (e.g. the
+        // library sweep) taking the key over — that fill owns the outcome.
+        guard !error.isCancellationError, let self, generation == fillGeneration else {
+          return
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply swift-paperless/ViewModel/DocumentListViewModel.swift]: 🤖 _Posted by Claude on Paul's behalf._

Confirmed, fixed in 1c2d36d9.

Traced through the code: `fillLibrary` calls `fillQuery` for the default list and every saved view. `fillQuery` runs `drainFill`, which cancels the list's task, and the list's `awaitCompletion` then threw a `CancellationError` that `watchCompletion` dropped. Page 1 of the replacement clears `filled_at`, so if it then failed, the list had rows, no error and an incomplete cache: no banner. An empty prefix with a recorded total stayed on placeholders.

**Fix: the list now follows the replacement instead of dropping it.**
- `DataModel`: new `DocumentListFillTracking` holds the decision. A fill the list moved on from reports nothing. A failed fill records its error. A cancellation the list didn't cause, meaning its generation is still current, means another fill took the query over, so the list follows it.
- `CachingBackend.waitForQueryWriters(_:)` waits until nothing owns the key in `activeFills`. It follows successive owners, which works because a drained owner's successor registers in the same main-actor job that retracts it, so there's no gap. `DocumentStore` passes it through.
- While following, the list counts as fetching: placeholders over an empty prefix, and the banner stays hidden. Once the key settles, the list reads `QueryStatus` once. The observation lags the last write, so it doesn't rely on that. If the cache still isn't complete, the list shows the failure state with Retry: the incomplete banner, or "couldn't load" over an empty prefix. There's no error text, because the error belonged to the other fill.
- Retry, a query switch and teardown reset this state, and a late follower can't overwrite it.

I didn't use `!isCacheComplete` as the failure signal while the replacement is still running. That would flag a healthy library sweep as incomplete for its whole run, and Retry would then cancel the sweep.

**Verified:**
- `swift test --package-path DataModel` passes (253 tests), including new `DocumentListFillTrackingTests`. They cover the follow-up decision and that a replacement that stopped short reaches `DocumentListState` as incomplete or unavailable, not loading.
- `swift test --package-path Persistence` passes.
- The app builds for the iOS Simulator.

**Inferred, not tested:** the view-model and `CachingRepository` wiring. AppShared has no test target, and I didn't reproduce a sweep taking the key over on a device or simulator.
